dlib/dnn/utilities_abstract.h

```diff
@@ -120,6 +120,20 @@ namespace dlib
 
 // ----------------------------------------------------------------------------------------
 
+    template <typename net_type>
+    inline size_t count_parameters(
+        const net_type& net
+    );
+    /*!
+        requires
+            - net_type is an object of type add_layer, add_loss_layer, add_skip_layer, or
+              add_tag_layer.
+        ensures
+            - This function returns the number of parameters of net if it has been properly
```

Comment:
How about, "this function returns the number of allocated parameters in the net.  E.g. if the network has not been trained then since nothing has been allocated yet it will return 0." instead.

Reply:
Oh, that's sounds better. I will submit a PR later if you don't do it before (I don't have my computer with me right now)

Reply:
Too late, thank you :)